ledger/metrics/src/test/lib/scala/com/daml/metrics/api/testing/InMemoryMetricsFactory.scala

```diff
@@ -65,35 +65,34 @@ object InMemoryMetricsFactory extends InMemoryMetricsFactory {
   }
 
   case class InMemoryTimer(initialContext: MetricsContext) extends Timer {
-    val runTimers: collection.concurrent.Map[MetricsContext, AtomicInteger] = TrieMap()
+    val data = InMemoryHistogram(initialContext)
 
     override def name: String = "test"
 
-    override def update(duration: Long, unit: TimeUnit)(implicit context: MetricsContext): Unit =
-      incrementForContext(context)
+    override def update(duration: Long, unit: TimeUnit)(implicit context: MetricsContext): Unit = {
+      data.update(TimeUnit.MILLISECONDS.convert(duration, unit))
+    }
```

Comment:
Nit: Just to be consistent with the other `update` definition:
```suggestion
    override def update(duration: Long, unit: TimeUnit)(implicit context: MetricsContext): Unit =
      data.update(TimeUnit.MILLISECONDS.convert(duration, unit))
```